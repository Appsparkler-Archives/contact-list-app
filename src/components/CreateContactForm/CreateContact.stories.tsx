import type { Meta, StoryObj } from "@storybook/react";
import { CreateContactForm } from "./CreateContactForm";

const meta = {
  title: "Components/Contact Form/Create Contact Form",
  component: CreateContactForm,
  tags: ["app"],
} satisfies Meta<typeof CreateContactForm>;

export default meta;

type Story = StoryObj<typeof meta>;

export const createContactForm: Story = {
  args: {},
};
