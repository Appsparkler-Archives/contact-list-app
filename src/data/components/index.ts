export * from "./standard-select-field";
