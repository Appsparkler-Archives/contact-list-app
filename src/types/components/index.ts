export * from "./StandardSelectField";
export * from "./FilterChip";
export * from "./CreateContactForm";
export * from "./ContactListAccordion";
export * from "./ContactForm";
export * from "./DeleteContact";
export * from "./FullScreenModalTransition";
export * from "./ViewContactModal";
