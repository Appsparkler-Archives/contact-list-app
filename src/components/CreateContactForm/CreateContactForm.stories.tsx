import type { Meta, StoryObj } from "@storybook/react";
import { CreateContactForm } from "./CreateContactForm";

const meta = {
  title: "Components/Create Contact Form",
  component: CreateContactForm,
  tags: ["components"],
} satisfies Meta<typeof CreateContactForm>;

export default meta;

type Story = StoryObj<typeof meta>;

export const createContactForm: Story = {
  args: {
    contact: {
      gender: "male",
      mobile: "+91-719-8101-355",
      name: "John Smith",
      type: "business",
      address: "Fathimapur",
      company: "PS",
      email: "appsparkler@gmail.com",
    },
  },
};
