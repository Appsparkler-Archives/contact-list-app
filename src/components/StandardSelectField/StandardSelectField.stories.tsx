import type { Meta, StoryObj } from "@storybook/react";
import { StandardSelectField } from "./StandardSelectField";
import { standardSelectFieldOptions } from "data";

const meta = {
  title: "Components/Standard Select Field",
  component: StandardSelectField,
  tags: ["components"],
} satisfies Meta<typeof StandardSelectField>;

export default meta;

type Story = StoryObj<typeof meta>;

export const standardSelectField: Story = {
  args: {
    id: "standard-select-field-id",
    label: "Gender",
    options: standardSelectFieldOptions,
    selectedOption: standardSelectFieldOptions[0].value,
  },
};
