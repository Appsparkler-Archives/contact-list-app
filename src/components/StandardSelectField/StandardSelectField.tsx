import { TSelectChangeEventHandler, TStandardSelectFieldFC } from "types";
import InputLabel from "@mui/material/InputLabel";
import MenuItem from "@mui/material/MenuItem";
import FormControl from "@mui/material/FormControl";
import Select from "@mui/material/Select";
import { useCallback } from "react";

export const StandardSelectField: TStandardSelectFieldFC = ({
  selectedOption,
  onChange,
  options,
  label,
  name,
  id,
}) => {
  const handleChange = useCallback<TSelectChangeEventHandler>(
    ({ target: { value, name } }) => {
      onChange(name, value);
    },
    [onChange]
  );
  return (
    <FormControl variant="standard" fullWidth>
      <InputLabel required id={id}>
        {label}
      </InputLabel>
      <Select<string>
        labelId={id}
        onChange={handleChange}
        label={label}
        required
        name={name}
        defaultValue={selectedOption}
      >
        {options.map((option) => (
          <MenuItem key={option.value} value={option.value}>
            {option.name}
          </MenuItem>
        ))}
      </Select>
    </FormControl>
  );
};
