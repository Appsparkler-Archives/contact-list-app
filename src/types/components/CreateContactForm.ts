import React from "react";
import { IContactFormData } from "types/data";

export type TCreateContactFormFC = React.FC<{
  onCreate: (contact: IContactFormData) => void;
  onCancel: () => void;
  //
  contact: IContactFormData;
}>;
