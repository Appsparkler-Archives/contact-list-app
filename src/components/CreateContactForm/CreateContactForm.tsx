import {
  AppBar,
  Box,
  Button,
  Dialog,
  Fab,
  IconButton,
  Slide,
  TextField,
  Toolbar,
  Typography,
} from "@mui/material";
import React, { useCallback, useState } from "react";
import { TCreateContactFormFC } from "types/components/CreateContactForm";
import AddIcon from "@mui/icons-material/Add";
import CloseIcon from "@mui/icons-material/Close";
import { TransitionProps } from "@mui/material/transitions";
import { IContactFormData, IStandardSelectFieldChangeHandler } from "types";
import { StandardSelectField } from "components/StandardSelectField/StandardSelectField";

const Transition = React.forwardRef(function Transition(
  props: TransitionProps & {
    children: React.ReactElement;
  },
  ref: React.Ref<unknown>
) {
  return <Slide direction="up" ref={ref} {...props} />;
});

export const CreateContactForm: TCreateContactFormFC = ({
  contact,
  onCreate,
  onCancel,
}) => {
  const [contactData, setContactData] = useState<IContactFormData>(contact);
  const [open, setOpen] = React.useState(false);

  const handleClickOpen = () => {
    setOpen(true);
  };

  const handleClose = () => {
    setOpen(false);
  };

  const handleSubmit = useCallback<React.FormEventHandler<HTMLFormElement>>(
    (event) => {
      event.preventDefault();
      onCreate(contactData);
      handleClose();
    },
    [contactData, onCreate]
  );

  const handleCancel = useCallback<
    React.MouseEventHandler<HTMLButtonElement>
  >(() => {
    onCancel();
  }, [onCancel]);

  const handleChangeTextField = useCallback<
    React.ChangeEventHandler<HTMLInputElement>
  >(({ target: { name, value } }) => {
    setContactData((oldContactData) => ({
      ...oldContactData,
      [name]: value,
    }));
  }, []);

  const handleChangeSelectField =
    useCallback<IStandardSelectFieldChangeHandler>((name, value) => {
      setContactData((oldContactData) => ({
        ...oldContactData,
        [name]: value,
      }));
    }, []);

  return (
    <React.Fragment>
      <Fab color="primary" aria-label="add" onClick={handleClickOpen}>
        <AddIcon />
      </Fab>
      <Dialog
        fullScreen
        open={open}
        onClose={handleClose}
        TransitionComponent={Transition}
      >
        <form onSubmit={handleSubmit}>
          <AppBar sx={{ position: "relative" }}>
            <Toolbar>
              <IconButton
                edge="start"
                color="inherit"
                onClick={handleClose}
                aria-label="close"
              >
                <CloseIcon />
              </IconButton>
              <Typography sx={{ ml: 2, flex: 1 }} variant="h6" component="div">
                Create Contact
              </Typography>
              <Button autoFocus color="inherit" type="submit">
                create
              </Button>
            </Toolbar>
          </AppBar>
          {/* Create Contact Form Fields */}
          <Box gap={2} p={2} flexDirection={"column"} display={"flex"}>
            <TextField
              fullWidth
              label="Name"
              variant="standard"
              name="name"
              value={contactData.name}
              onChange={handleChangeTextField}
              required
            />
            <TextField
              fullWidth
              label="Mobile"
              variant="standard"
              name="mobile"
              value={contactData.mobile}
              onChange={handleChangeTextField}
              required
            />
            <TextField
              fullWidth
              label="Email"
              variant="standard"
              value={contactData.email}
              onChange={handleChangeTextField}
              name="email"
            />
            <TextField
              fullWidth
              label="Address"
              variant="standard"
              value={contactData.address}
              onChange={handleChangeTextField}
              name="address"
            />
            <StandardSelectField
              options={[
                { name: "Female", value: "female" },
                { name: "Male", value: "male" },
              ]}
              id="contact-field--gender"
              label="Gender"
              onChange={handleChangeSelectField}
              selectedOption={contactData.gender}
              name="gender"
            />
            <StandardSelectField
              options={[
                { name: "Personal", value: "personal" },
                { name: "Business", value: "business" },
              ]}
              id="contact-field--type"
              label="Type"
              onChange={handleChangeSelectField}
              selectedOption={contactData.type}
              name="type"
            />
            <TextField
              fullWidth
              label="Company"
              variant="standard"
              value={contactData.company}
              name="company"
              onChange={handleChangeTextField}
            />
            <Box
              display={"flex"}
              flexDirection={"row"}
              justifyContent={"space-around"}
              mt={3}
            >
              <Button
                type="button"
                size="large"
                variant="outlined"
                onClick={handleCancel}
              >
                Cancel
              </Button>

              <Button variant="contained" size="large" type="submit">
                Create
              </Button>
            </Box>
          </Box>
        </form>
      </Dialog>
    </React.Fragment>
  );
};
