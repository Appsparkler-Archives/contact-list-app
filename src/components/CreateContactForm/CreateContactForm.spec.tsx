import {
  AppBar,
  Box,
  Button,
  Dialog,
  Fab,
  IconButton,
  Slide,
  TextField,
  Toolbar,
  Typography,
} from "@mui/material";
import React, { useCallback, useEffect, useState } from "react";
import { TCreateContactFormFC } from "types/components/CreateContactForm";
import AddIcon from "@mui/icons-material/Add";
import CloseIcon from "@mui/icons-material/Close";
import { TransitionProps } from "@mui/material/transitions";
import { IContactFormData, IStandardSelectFieldChangeHandler } from "types";
import { StandardSelectField } from "components/StandardSelectField/StandardSelectField";
import { fireEvent, render, screen } from "@testing-library/react";
import renderer, { act } from "react-test-renderer";

const CreateContactFormOpened = () => {
  const contact: IContactFormData = {
    name: "",
    mobile: "",
    gender: "male",
    type: "personal",
  };

  const handleMount = useCallback(async () => {
    // eslint-disable-next-line
    const addButton = screen.getByRole("button", { name: /add/ });
    if (addButton) {
      expect(addButton).toBeTruthy();
      // eslint-disable-next-line testing-library/no-unnecessary-act
      await act(() => {
        fireEvent.click(addButton);
      });
    }
  }, []);

  useEffect(() => {
    handleMount();
  }, [handleMount]);

  return (
    <CreateContactForm
      onCreate={function (contact: IContactFormData): void {
        throw new Error("Function not implemented.");
      }}
      onCancel={function (): void {
        throw new Error("Function not implemented.");
      }}
      contact={contact}
    />
  );
};

describe("testing renderer", () => {
  test("abc", async () => {
    const { update } = renderer.create(<div>Hello</div>);
  });
});

const Transition = React.forwardRef(function Transition(
  props: TransitionProps & {
    children: React.ReactElement;
  },
  ref: React.Ref<unknown>
) {
  return <Slide direction="up" ref={ref} {...props} />;
});

const CreateContactForm: TCreateContactFormFC = ({
  contact,
  onCreate,
  onCancel,
}) => {
  const [contactData, setContactData] = useState<IContactFormData>(contact);
  const [open, setOpen] = React.useState(false);

  const handleClickOpen = () => {
    setOpen(true);
  };

  const handleClose = () => {
    setOpen(false);
  };

  const handleSubmit = useCallback<React.FormEventHandler<HTMLFormElement>>(
    (event) => {
      event.preventDefault();
      onCreate(contactData);
    },
    [contactData, onCreate]
  );

  const handleCancel = useCallback<
    React.MouseEventHandler<HTMLButtonElement>
  >(() => {
    onCancel();
  }, [onCancel]);

  const handleChangeTextField = useCallback<
    React.ChangeEventHandler<HTMLInputElement>
  >(({ target: { name, value } }) => {
    setContactData((oldContactData) => ({
      ...oldContactData,
      [name]: value,
    }));
  }, []);

  const handleChangeSelectField =
    useCallback<IStandardSelectFieldChangeHandler>((name, value) => {
      setContactData((oldContactData) => ({
        ...oldContactData,
        [name]: value,
      }));
    }, []);

  return (
    <React.Fragment>
      <Fab color="primary" aria-label="add" onClick={handleClickOpen}>
        <AddIcon />
      </Fab>
      <Dialog
        fullScreen
        open={open}
        onClose={handleClose}
        TransitionComponent={Transition}
      >
        <AppBar sx={{ position: "relative" }}>
          <Toolbar>
            <IconButton
              edge="start"
              color="inherit"
              onClick={handleClose}
              aria-label="close"
            >
              <CloseIcon />
            </IconButton>
            <Typography sx={{ ml: 2, flex: 1 }} variant="h6" component="div">
              Create Contact
            </Typography>
            <Button autoFocus color="inherit" onClick={handleClose}>
              save
            </Button>
          </Toolbar>
        </AppBar>
        {/* Create Contact Form Fields */}
        <Box gap={2} flexDirection={"column"} display={"flex"}>
          <TextField
            fullWidth
            label="Name"
            variant="standard"
            name="name"
            value={contactData.name}
            onChange={handleChangeTextField}
            required
          />
          <TextField
            fullWidth
            label="Mobile"
            variant="standard"
            name="mobile"
            value={contactData.mobile}
            onChange={handleChangeTextField}
            required
          />
          <TextField
            fullWidth
            label="Email"
            variant="standard"
            value={contactData.email}
            onChange={handleChangeTextField}
            name="email"
          />
          <TextField
            fullWidth
            label="Address"
            variant="standard"
            value={contactData.address}
            onChange={handleChangeTextField}
            name="address"
          />
          <StandardSelectField
            options={[
              { name: "Female", value: "female" },
              { name: "Male", value: "male" },
            ]}
            id="contact-field--gender"
            label="Gender"
            onChange={handleChangeSelectField}
            selectedOption={contactData.gender}
            name="gender"
          />
          <StandardSelectField
            options={[
              { name: "Personal", value: "personal" },
              { name: "Business", value: "business" },
            ]}
            id="contact-field--type"
            label="Type"
            onChange={handleChangeSelectField}
            selectedOption={contactData.type}
            name="type"
          />
          <TextField
            fullWidth
            label="Company"
            variant="standard"
            value={contactData.company}
            name="company"
            onChange={handleChangeTextField}
          />
          <Box
            display={"flex"}
            flexDirection={"row"}
            justifyContent={"space-around"}
            mt={3}
          >
            <Button
              type="button"
              size="large"
              variant="outlined"
              onClick={handleCancel}
            >
              Cancel
            </Button>

            <Button variant="contained" size="large" type="submit">
              Create
            </Button>
          </Box>
        </Box>
      </Dialog>
    </React.Fragment>
  );
};

describe("CreateContactForm - snapshots", () => {
  const contact: IContactFormData = {
    name: "",
    mobile: "",
    email: "",
    address: "",
    gender: "female",
    type: "personal",
    company: "",
  };

  const props = {
    contact: contact,
    onCreate: jest.fn(),
    onCancel: jest.fn(),
  };

  it("closed modal snapsnhot", () => {
    const { baseElement } = render(<CreateContactForm {...props} />);
    expect(baseElement).toMatchSnapshot();
  });

  it("open modal snapsnhot", async () => {
    const { baseElement } = render(<CreateContactForm {...props} />);
    const addButton = screen.getByRole("button", { name: /add/ });
    expect(addButton).toBeTruthy();
    fireEvent.click(addButton);
    expect(baseElement).toMatchSnapshot();
  });
});
