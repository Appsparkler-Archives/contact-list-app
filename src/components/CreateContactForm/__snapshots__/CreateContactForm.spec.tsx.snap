// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateContactForm - snapshots closed modal snapsnhot 1`] = `
<body>
  <div>
    <button
      aria-label="add"
      class="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeLarge MuiFab-primary MuiFab-root MuiFab-circular MuiFab-sizeLarge MuiFab-primary css-1fo2sjy-MuiButtonBase-root-MuiFab-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="AddIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</body>
`;

exports[`CreateContactForm - snapshots open modal snapsnhot 1`] = `
<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  >
    <button
      aria-label="add"
      class="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeLarge MuiFab-primary MuiFab-root MuiFab-circular MuiFab-sizeLarge MuiFab-primary css-1fo2sjy-MuiButtonBase-root-MuiFab-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="AddIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-testid="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
      role="presentation"
      style="webkit-transform: none; transform: none; webkit-transition: -webkit-transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        aria-labelledby=":r1:"
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiDialog-paperFullScreen css-m9glnp-MuiPaper-root-MuiDialog-paper"
        role="dialog"
      >
        <form>
          <header
            class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-10ghrmp-MuiPaper-root-MuiAppBar-root"
            style="padding-right: 1024px;"
          >
            <div
              class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
            >
              <button
                aria-label="close"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeMedium css-1d6w9lk-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="CloseIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <div
                class="MuiTypography-root MuiTypography-h6 css-6b3s0t-MuiTypography-root"
              >
                Create Contact
              </div>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit Mui-focusVisible MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit css-1y942vo-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="submit"
              >
                create
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                >
                  <span
                    class="css-y4cjyz-MuiTouchRipple-ripple MuiTouchRipple-ripple MuiTouchRipple-rippleVisible MuiTouchRipple-ripplePulsate"
                    style="width: 1px; height: 1px; top: -0.5px; left: -0.5px;"
                  >
                    <span
                      class="MuiTouchRipple-child MuiTouchRipple-childPulsate"
                    />
                  </span>
                </span>
              </button>
            </div>
          </header>
          <div
            class="MuiBox-root css-a9s4rr"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for=":r2:"
                id=":r2:-label"
              >
                Name
                <span
                  aria-hidden="true"
                  class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-l4u8b9-MuiInputBase-root-MuiInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                  id=":r2:"
                  name="name"
                  required=""
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for=":r3:"
                id=":r3:-label"
              >
                Mobile
                <span
                  aria-hidden="true"
                  class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-l4u8b9-MuiInputBase-root-MuiInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                  id=":r3:"
                  name="mobile"
                  required=""
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for=":r4:"
                id=":r4:-label"
              >
                Email
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-l4u8b9-MuiInputBase-root-MuiInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                  id=":r4:"
                  name="email"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for=":r5:"
                id=":r5:-label"
              >
                Address
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-l4u8b9-MuiInputBase-root-MuiInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                  id=":r5:"
                  name="address"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="true"
                id="contact-field--gender"
              >
                Gender
                <span
                  aria-hidden="true"
                  class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl  css-m5hdmq-MuiInputBase-root-MuiInput-root-MuiSelect-root"
                required=""
              >
                <div
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-labelledby="contact-field--gender mui-component-select-gender"
                  class="MuiSelect-select MuiSelect-standard MuiInputBase-input MuiInput-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
                  id="mui-component-select-gender"
                  role="button"
                  tabindex="0"
                >
                  Female
                </div>
                <input
                  aria-hidden="true"
                  aria-invalid="false"
                  class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                  name="gender"
                  required=""
                  tabindex="-1"
                  value="female"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="true"
                id="contact-field--type"
              >
                Type
                <span
                  aria-hidden="true"
                  class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl  css-m5hdmq-MuiInputBase-root-MuiInput-root-MuiSelect-root"
                required=""
              >
                <div
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-labelledby="contact-field--type mui-component-select-type"
                  class="MuiSelect-select MuiSelect-standard MuiInputBase-input MuiInput-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
                  id="mui-component-select-type"
                  role="button"
                  tabindex="0"
                >
                  Personal
                </div>
                <input
                  aria-hidden="true"
                  aria-invalid="false"
                  class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                  name="type"
                  required=""
                  tabindex="-1"
                  value="personal"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for=":r6:"
                id=":r6:-label"
              >
                Company
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-l4u8b9-MuiInputBase-root-MuiInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                  id=":r6:"
                  name="company"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="MuiBox-root css-vxb414"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge css-9y1egq-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Cancel
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge css-jh47zj-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="submit"
              >
                Create
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
    <div
      data-testid="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
